pennylane/ops/functions/map_wires.py

```diff
@@ -28,8 +27,32 @@
 from pennylane.workflow import QNode
 
 
+@overload
 def map_wires(
-    input: Union[Operator, MeasurementProcess, QuantumScript, QNode, Callable],
+    input: Operator, wire_map: dict, queue: bool = False, replace: bool = False
+) -> Operator: ...
+@overload
+def map_wires(
+    input: MeasurementProcess, wire_map: dict, queue: bool = False, replace: bool = False
+) -> MeasurementProcess: ...
+@overload
+def map_wires(
+    input: QuantumScript, wire_map: dict, queue: bool = False, replace: bool = False
+) -> tuple[QuantumScriptBatch, PostprocessingFn]: ...
+@overload
+def map_wires(
+    input: QNode, wire_map: dict, queue: bool = False, replace: bool = False
+) -> QNode: ...
+@overload
+def map_wires(
+    input: Callable, wire_map: dict, queue: bool = False, replace: bool = False
+) -> Callable: ...
+@overload
+def map_wires(
+    input: QuantumScriptBatch, wire_map: dict, queue: bool = False, replace: bool = False
+) -> tuple[QuantumScriptBatch, PostprocessingFn]: ...
+def map_wires(
+    input: Union[Operator, MeasurementProcess, QuantumScript, QNode, Callable, QuantumScriptBatch],
     wire_map: dict,
     queue=False,
     replace=False,
```

Comment:
Just out of curiosity, there is nothing preventing the combination `queue=False, replace=True`, as far as I can see, but the docstring sounds like `replace=True` is only doing something when `queue=True`.

Reply:
Not really sure actually, but probably out of scope of this change.